Added mailer method to actually send an email on submission of a volunteer application

JessSumner opened the pull request:
Why:
This is so the rescue can immediately respond to any new volunteer applications.

How:
This mailer follows the same logic as the adoption application mailer.

JessSumner committed:
Ammended the radio buttons to be linked with the volunteer application object

Why:
The values checked on the radio buttons were not being saved to the database.

This commit:
Changes radio_button_tag to f.radio_button. This links the radio button with f (the form object) and therefore the radio button value is saved to the database.

JessSumner committed:
New Volunteer Application Mailer Generated

Mailer has been generated.
It shares the admin_email method with the new adoption application mailer so this method has been moved to the application mailer file.
Created a constant FORM_FIELDS so that I can use it to create the html table in the mailer.
Created a preview for both html and text.
Updated the params in the controller as some options had been added to the migration but the controller had not been updated.

JessSumner committed:
Added method to actually send email

On saving a volunteer application an email will be generated.

JessSumner committed:
Added submission time of the form

Having a submission time on the form would help the rescue keep track of when they received it so they can follow up promptly.

Defined a method in app/helper that converts the created_at value to local time and then formats it to be easier for the reader.
Had to add_template_helper to the Mailer to access these methods.

JessSumner committed:
Removed trailing whitespace from mailer and helper

JessSumner committed:
Removed extra line from app/helper file

MattMSumner commented on app/controllers/volunteer_applications_controller.rb:
```diff
@@ -7,6 +7,8 @@ def create
     @volunteer_application =
       VolunteerApplication.new(volunteer_application_params)
     if @volunteer_application.save
+      NewVolunteerApplicationMailer.volunteer_notification(
```
How do you feel about breaking up this line differently? i.e.

``` rb
NewVolunteerApplicationMailer.
  volunteer_notification(@volunteer_application).
  deliver_now
```

JessSumner replied:
So I have changed my code as suggested as it does read better. The thing that threw me is the syntax highlighting changes. So NewVolunteerApplicationMailer no longer highlights as a class. I have submitted a form and the mailer went out so it works, but it deterred me initially.

MattMSumner commented on app/mailers/new_volunteer_application_mailer.rb:
```diff
@@ -0,0 +1,9 @@
+
+  def volunteer_notification(volunteer_application)
+    @volunteer_application = volunteer_application
+    mail(to: admin_email,
```
How do you feel about giving each key pair it's own line? i.e.

``` rb
mail(
  to: admin_email,
  subject: "A new volunteer application has been submitted"
)
```

MattMSumner commented on app/views/new_volunteer_application_mailer/volunteer_notification.html.erb:
```diff
@@ -0,0 +1,26 @@
+<!DOCTYPE html>
```
Should this be here?

MattMSumner replied:
I guess it should because it's a mailer

MattMSumner commented on app/views/volunteer_applications/_volunteer_form_fields.html.erb:
```diff
@@ -22,74 +22,78 @@
   <%= f.text_field :email %>
 
-  <%= f.label :reason_for_volunteering, "Why do you want to quarantine or 
+  <%= f.label :reason_for_volunteering, "Why do you want to quarantine or
```
How would you feel about putting each argument on it's own line? i.e.

``` erb
<%= f.label :reason_for_volunteering,
  "Why do you want to quarantine or foster?" %>
```

MattMSumner commented on app/views/volunteer_applications/_volunteer_form_fields.html.erb:
```diff
@@ -22,74 +22,78 @@
-    <%= label_tag(:volunteering_type_foster, "I would like to be a foster home"
-        ) %></br>
+    <%= f.radio_button :volunteering_type, "foster" %>
+    <%= f.label :volunteering_type, "I would like to be a foster home",
```
How would you feel about putting each argument on it's own line?

MattMSumner commented on app/views/volunteer_applications/_volunteer_form_fields.html.erb:
```diff
@@ -22,74 +22,78 @@
-    <%= label_tag(:volunteering_type_quarantine, "I would like to be a 
-                                                  quarantine home") %></br>
+    <%= f.radio_button :volunteering_type, "quarantine" %>
+    <%= f.label :volunteering_type, "I would like to be a quarantine home",
```
Same here?

MattMSumner commented on app/views/volunteer_applications/_volunteer_form_fields.html.erb:
```diff
@@ -22,74 +22,78 @@
-    <%= label_tag(:volunteering_type_either, "I would like to foster or 
-                                              quarantine") %></br>
+    <%= f.radio_button :volunteering_type, "either", checked: true %>
+    <%= f.label :volunteering_type_either, "I would like to foster or
```
and here:

``` erb
<%= f.label :volunteering_type_either,
  "I would like to foster or quarantine",
  value: "either" %></br>
```

MattMSumner commented on app/views/volunteer_applications/_volunteer_form_fields.html.erb:
```diff
@@ -22,74 +22,78 @@
-    <%= radio_button_tag(:vet_appointments, "yes", true) %>
-    <%= label_tag(:vet_appointments_yes, "I can keep vet appointments") %></br>
+    <%= f.radio_button :vet_appointments, true, checked: true %>
+    <%= f.label :vet_appointments, "I can keep vet appointments",
```
How would you feel about 1 line per argument?

MattMSumner commented on app/views/volunteer_applications/_volunteer_form_fields.html.erb:
```diff
@@ -22,74 +22,78 @@
-    <%= radio_button_tag(:vet_appointments, "no") %>
-    <%= label_tag(:vet_appointments_no, "I can not keep vet appointments") %>
+    <%= f.radio_button :vet_appointments, false %>
+    <%= f.label :vet_appointments, "I can not keep vet appointments",
```
How would you feel about 1 line per argument?

MattMSumner commented on app/views/volunteer_applications/_volunteer_form_fields.html.erb:
```diff
@@ -22,74 +22,78 @@
-    <%= label_tag(:caring_for_ill_rats_yes, "I am comfortable caring for sick 
-                                            or injured rats.") %></br>
+    <%= f.radio_button :caring_for_ill_rats, true, checked: true %>
+    <%= f.label :caring_for_ill_rats, "I am comfortable caring for sick or 
```
How would you feel about 1 line per argument?

MattMSumner commented on app/views/volunteer_applications/_volunteer_form_fields.html.erb:
```diff
@@ -22,74 +22,78 @@
-    <%= label_tag(:caring_for_ill_rats_no, "I can not look after sick or 
-                                            injured rats.") %></br>
+    <%= f.radio_button :caring_for_ill_rats, false %>
+    <%= f.label :caring_for_ill_rats, "I can not look after sick or injured
```
How would you feel about 1 line per argument?

MattMSumner commented on app/views/volunteer_applications/_volunteer_form_fields.html.erb:
```diff
@@ -102,17 +106,19 @@
-    <%= radio_button_tag(:other_animals, "yes") %>
-    <%= label_tag(:other_animals_yes, "I have other animals in my home.") %>
+    <%= f.radio_button :other_animals, true %>
+    <%= f.label :other_animals, "I have other animals in my home.",
```
How would you feel about 1 line per argument?

MattMSumner commented on app/views/volunteer_applications/_volunteer_form_fields.html.erb:
```diff
@@ -102,17 +106,19 @@
-    <%= label_tag(:other_animals_no, "I do not have any other animals in my 
-                                      home.") %></br>
+    <%= f.radio_button :other_animals, false, checked: true %>
+    <%= f.label :other_animals, "I do not have any other animals in my home.",
```
How would you feel about 1 line per argument?

MattMSumner commented on app/views/volunteer_applications/_volunteer_form_fields.html.erb:
```diff
@@ -122,18 +128,18 @@
 
-  <%= f.label :hours_away_from_home, "How long are you away from home per 
-                                      day?" %>
+  <%= f.label :hours_away_from_home, "How long are you away from home per day?"
```
How would you feel about 1 line per argument?

MattMSumner commented on app/views/volunteer_applications/_volunteer_form_fields.html.erb:
```diff
@@ -122,18 +128,18 @@
   <%= f.text_field :hours_away_from_home %>
   
-  <%= f.label :rat_interaction_time, "Will you be able to spend at least an 
+  <%= f.label :rat_interaction_time, "Will you be able to spend at least an
```
How would you feel about 1 line per argument?

MattMSumner commented on app/views/volunteer_applications/_volunteer_form_fields.html.erb:
```diff
@@ -153,15 +159,14 @@
-                                              adopters for the rats I am caring
-                                              for." %></br>
+    <%= f.radio_button :adoption_appointments, true, checked: true %>
+    <%= f.label :adoption_appointments, "I will be able to meet with adopters
```
How would you feel about 1 line per argument?

MattMSumner commented on app/views/volunteer_applications/_volunteer_form_fields.html.erb:
```diff
@@ -153,15 +159,14 @@
+                                         for the rats I am caring for." %></br>
+
+    <%= f.radio_button :adoption_appointments, false %>
+    <%= f.label :adoption_appointments, "I will not be able to meet with 
```
How would you feel about 1 line per argument?

MattMSumner commented on test/mailers/previews/new_volunteer_application_mailer_preview.rb:
```diff
@@ -0,0 +1,8 @@
+# Preview all emails at http://localhost:3000/rails/mailers/new_volunteer_application_mailer
```
How would you feel about removing the comments from this file?

JessSumner replied:
I know it's best to cut out comments. I would like to keep this for the url. I'm not sure I'll remember where to go otherwise.

MattMSumner commented:
Just some styling comments, otherwise this LGTM

JessSumner committed:
Some styling changes to the volunteer form and mailer.

Adjusted where the code was broken up across multiple lines. Split up the code so each argument is on its own line rather than breaking within an argument.
Removed a comment from the mailer preview. Although kept one for reference.
